doc/InfiniTimeVision.md

```diff
@@ -0,0 +1,32 @@
+# InfiniTime Vision
+
+The purpose of this document is to steer efforts towards a common goal, and as such should be taken into consideration with all developments.
+
+## InfiniTime
+
+InfiniTime is a community-built smartwatch firmware, providing freedom and privacy that's usually insufficient in wearable technology.
+
+InfiniTime is not to be used for medical or other health tracking purposes.
+
+## Core Principles
+
+- KISS (Keep It Short and Simple)
+- Reliability
+- Battery efficiency
+- Easy and simple navigation
+- Behaviour should be predictable and easy to understand
+- Prefer solid default experience over customization
+- Personalization is achieved through custom watchfaces.
+  More options may be available through a companion app.
+- Use standard protocols and methods
```

Comment:
Could we be more specific? For example BLE services with a link to a summary of the available ones. Maybe also explain the rational that justifies the burden of not just rolling out our custom easy-to-implement/debug solutions?

Reply:
One of the goals of implementing standard and widely used protocols is to avoid re-inventing the wheel. Why spend time and efforts designing a protocol when we can just implement an existing one that fits our needs? Designing a protocol like DFU or file transfer protocol is not that easy. By using a well established one that is already used for similar use-case, we ensure that this protocol will probably work fine for us too.

Also, it makes the integration with other devices and apps a lot more easier. If a companion app already implements such a protocol, adding support for the PineTime will be virtually free. Why should we require companion apps developers to implement yet another protocol to set the time or to fetch heart rate values when they have already implement them multiple times?

In InfiniTime, we implement standard protocols from the BLE spec (GATT services : https://github.com/oesmith/gatt-xml).
For the BLE FS API, we implemented the BLE File Transfer Protocol from Adafruit (https://github.com/adafruit/Adafruit_CircuitPython_BLE_File_Transfer).
On the other hand, we couldn't find anything standard for the motion service, so we designed a custom one (https://github.com/InfiniTimeOrg/InfiniTime/blob/develop/doc/MotionService.md)

We choose the protocol on a case-by-case basis, but designing and implementing a new one from scratch should always be considered as a last resort.